components/ee/agent-smith/pkg/classifier/sinature.go

```diff
@@ -213,22 +233,28 @@ func (s *Signature) matchELF(in io.ReaderAt) (bool, error) {
 
 // ExtractELFSymbols extracts all ELF symbol names from an ELF binary
 func ExtractELFSymbols(executable *elf.File) ([]string, error) {
-	var symbols []string
 	syms, err := executable.Symbols()
 	if err != nil && err != elf.ErrNoSymbols {
 		return nil, xerrors.Errorf("cannot get dynsym section: %w", err)
 	}
-	for _, s := range syms {
-		symbols = append(symbols, s.Name)
-	}
 
 	dynsyms, err := executable.DynamicSymbols()
 	if err != nil && err != elf.ErrNoSymbols {
 		return nil, xerrors.Errorf("cannot get dynsym section: %w", err)
 	}
+
+	symbols := make([]string, len(syms)+len(dynsyms))
+	i := 0
+	for _, s := range syms {
+		symbols[i] = s.Name
+		i += 1
+	}
+
 	for _, s := range dynsyms {
-		symbols = append(symbols, s.Name)
+		symbols[i] = s.Name
+		i += 1
```

Comment:
Are these refactoring?

Reply:
We know the number of elements, so I allocate the underlying array once instead of appending to the slice which includes resizing i.e. reallocation of the array.